src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedRootModuleType.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.CodeGen;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents the <see cref="Cci.RootModuleType"/> that will be emitted. Used for <see
+    /// cref="SynthesizedRootModuleTypeStaticConstructor.ContainingType"/> so that tests can use proper metadata names.
+    /// See <see cref="CompilationTestData.GetMethodsByName"/>.
+    /// </summary>
+    internal sealed class SynthesizedRootModuleType : NamedTypeSymbol
```

Comment:
>SynthesizedRootModuleType [](start = 26, length = 25)

Addition of this symbol feels wrong given its purpose stated in the doc comment above

Reply:
What is the alternative that you recommend? Special-cased logic somewhere else so that test code can still use `verifier.VerifyIL("<Module>..cctor", ...)`, or is your opinion that test code should change?

Reply:
> What is the alternative that you recommend? 

It feels like the IL is going to be very simple. I would be fine with just observing what the "initializer" is doing at runtime. Or we can come up with a way to verify IL without introducing confusing and unnecessary artifacts into the pipeline.

---
In reply to: [409487198](https://github.com/dotnet/roslyn/pull/43281#discussion_r409487198) [](ancestors = 409487198)

Reply:
@AlekseyTs I have tests that do `verifier.VerifyMemberInIL("<Module>..cctor", expected: false);` to assert that no empty static constructor is created. Does that seem like a valuable assertion to keep? If so, does that mean we'll have to come up with a new way to verify IL?

Reply:
>  have tests that do verifier.VerifyMemberInIL("<Module>..cctor", expected: false); to assert that no empty static constructor is created. Does that seem like a valuable assertion to keep?

I would prefer that we examine actual metadata that we emitted rather than something that we collected during emit process on the side. Can't we simply examine symbols in symbolValidator (in CompileAndVerify) that ```<Module>``` doesn't have a constructor? I think it will be among the types we import. If not, we can always examine metadata directly, I think CompileAndVerify has a hook for that too.

>  If so, does that mean we'll have to come up with a new way to verify IL?

As I mentioned earlier, I would be fine without verifying IL by dumping it. Observing expected behavior at runtime should be just fine, I think.


---
In reply to: [409712098](https://github.com/dotnet/roslyn/pull/43281#discussion_r409712098) [](ancestors = 409712098)